### PR TITLE
fix build_with_container.sh

### PR DESCRIPTION
**Pull request by algobot76**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

I can't run `make` with the current build script on mac as `docker` doesn't exist in `/etc/group`. So I tweak the `build_with_container.sh` script to make it compatible with mac.

**Which issue(s) this PR fixes**:

n/a

**Special notes for your reviewer**:

n/a

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/build_with_container.sh [hunk 1]
```diff
@@ -23,11 +23,12 @@ set -o pipefail
 KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 MOUNTPATH="${MOUNTPATH:-/kubeedge}"
 KUBEEDGE_BUILD_IMAGE=${KUBEEDGE_BUILD_IMAGE:-"kubeedge/build-tools"}
-DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}"
+USERNAME="$UID"
+DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}" && USERNAME="${UID}:${DOCKER_GID}"
```
Comment (gy95) [inline hack/make-rules/build_with_container.sh:27]: what's the difference before and after this fix? Isn't that exactly the same? :)

Comment (algobot76) [reply hack/make-rules/build_with_container.sh:27]: To prevent the script from terminating with exit code 1 when it can't find the gid of docker.

Comment (gy95) [reply hack/make-rules/build_with_container.sh:27]: on mac, there's no file `/etc/group`, and then `grep '^docker:' /etc/group | cut -f3 -d:)` cmd will fail. And by running `$?` cmd, we will find the exit code is not 0.
And on mac, by appending `&& USERNAME="${UID}:${DOCKER_GID}"`, the command will run successfully and the exit code will be 0
That's why this patch will fix on mac terminating with exit code 1

File: hack/make-rules/build_with_container.sh [hunk 2]
```diff
@@ -23,11 +23,12 @@ set -o pipefail
 KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 MOUNTPATH="${MOUNTPATH:-/kubeedge}"
 KUBEEDGE_BUILD_IMAGE=${KUBEEDGE_BUILD_IMAGE:-"kubeedge/build-tools"}
-DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}"
+UID_GID="$UID"
```
Comment (gy95) [inline hack/make-rules/build_with_container.sh:26]: UID="$UID" ?

Comment (algobot76) [reply hack/make-rules/build_with_container.sh:26]: image

By default, we use uid as the default parameter for `-u`. Then append the docker's gid to the uid in the form of `uid:gid` if it is present.

**Pull request closed without merging**